packages/editor/src/components/char-count/index.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * WordPress dependencies
+ */
+import { withSelect } from '@wordpress/data';
+import { _x } from '@wordpress/i18n';
+import { count as wordCount } from '@wordpress/wordcount';
+
+function CharCount( { content } ) {
+	/*
+	 * Only count the number of characters (no spaces).
+	 * Do not translate into your own language.
+	 */
+	const charCountType = _x( 'characters_excluding_spaces', 'Word count type. Do not translate!' );
+
+	return (
+		<span className="word-count">{ wordCount( content, charCountType ) }</span>
```

Comment:
`word-count` isn't the right class name here. Instead I think it should be called `char-count`, reflecting the component name. To get the styling to work an extra selector will have to be added here for the new `.char-count` class:
https://github.com/WordPress/gutenberg/blob/4fc05d35b710e4545c4a07e04d2f1e2bad692373/packages/editor/src/components/table-of-contents/style.scss#L33-L39

It is a bit unusual that these components (`CharCount`, `WordCount`) receive their style from a completely different component (`TableOfContentsPanel`), but I think it's outside of the scope of this PR to fix that.